etna/lib/etna/redirect.rb

```diff
@@ -0,0 +1,36 @@
+module Etna
+  def self.Redirect(req)
+    Redirect.new(req)
+  end
+
+  class Redirect
+    def initialize(request)
+      @request = request
+    end
+
+    def to(path, &block)
+      return Rack::Response.new(
+        [ { errors: [ 'Cannot redirect out of domain' ] }.to_json ], 422,
+        { 'Content-Type' => 'application/json' }
+      ).finish unless matches_domain?(path)
+
+      response = Rack::Response.new
+
+      response.redirect(path.gsub("http://", "https://"), 302)
+
+      yield response if block_given?
+
+      response.finish
+    end
+
+    private
+
+    def matches_domain?(path)
+      top_domain(@request.server_name) == top_domain(URI(path).host)
+    end
+
+    def top_domain(host_name)
+      host_name.split('.')[-2..-1].join('.')
```

Comment:
This does not seem to work on development, which is unfortunate?

```
Puma caught this error: undefined method `join' for nil:NilClass (NoMethodError)
/etna/lib/etna/redirect.rb:33:in `top_domain'
/etna/lib/etna/redirect.rb:29:in `matches_domain?'
/etna/lib/etna/redirect.rb:15:in `to'
/etna/lib/etna/auth.rb:68:in `fail_or_redirect'
/etna/lib/etna/auth.rb:19:in `call'
/etna/lib/etna/symbolize_params.rb:10:in `call'
/etna/lib/etna/parse_body.rb:38:in `call'
/etna/lib/etna/metrics.rb:22:in `call'
/etna/lib/etna/cross_origin.rb:34:in `actual_response'
/etna/lib/etna/cross_origin.rb:11:in `call'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/configuration.rb:238:in `call'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/server.rb:638:in `block in handle_request'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/thread_pool.rb:337:in `with_force_shutdown'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/server.rb:637:in `handle_request'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/server.rb:394:in `process_client'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/server.rb:250:in `block in run'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/thread_pool.rb:145:in `block in spawn_thread'
```

Trying to load Vulcan with no cookie set, from https://vulcan.development.local

Reply:
should be fixed?